internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -538,7 +538,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, xr), errUpdateStatus)
 	}
 	if rev := xr.GetCompositionRevisionReference(); rev != nil && (origRev == nil || *rev != *origRev) {
-		r.record.Event(xr, event.Normal(reasonResolve, "Selected composition revision: %s", rev.Name))
+		r.record.Event(xr, event.Normal(reasonResolve, fmt.Sprintf("Selected composition revision: %s", rev.Name)))
```

Comment:
Normal takes key/value pairs from the 3rd argument on, so the other option here is to make it
```suggestion
		r.record.Event(xr, event.Normal(reasonResolve, "Selected composition revision", "name", rev.Name)))
```

Reply:
Looking better at it, we don't use annotations much, just in a few places, these are not shown when viewing events, I'd say we can continue using `fmt.Sprintf`, wdyt? CC @negz @sttts